src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -461,53 +503,59 @@ private void ProcessParallelParameterSet()
 
             if (AsJob)
             {
+                // Add child task job.
                 var taskChildJob = new PSTaskChildJob(
                     Parallel,
                     _usingValuesMap,
                     InputObject);
 
                 _taskJob.AddJob(taskChildJob);
+
+                return;
             }
-            else
-            {
-                // Write any streaming data
-                _taskDataStreamWriter.WriteImmediate();
 
-                var task = new System.Management.Automation.PSTasks.PSTask(
-                     Parallel,
-                     _usingValuesMap,
-                     InputObject,
-                     _taskDataStreamWriter);
+            // Write any streaming data
+            _taskDataStreamWriter.WriteImmediate();
 
-                // Add task to task pool.
-                // Block if the pool is full and wait until task can be added.
-                _taskPool.Add(task, _taskDataStreamWriter);
+            // Add to task collection for processing.
+            if (_taskCollection.IsOpen)
+            {
+                try
+                {
+                    _taskCollection.Add(
+                        new System.Management.Automation.PSTasks.PSTask(
+                            Parallel,
+                            _usingValuesMap,
+                            InputObject,
+                            _taskDataStreamWriter));
+                }
+                catch (InvalidOperationException)
```

Comment:
What would the circumstances for an `InvalidOperationException` be? Should the user be warned if their task isn't run or handled properly?

Reply:
This exception is thrown by PSDataCollection<> if this method is used when the collection is closed, which is possible but unexpected.  I'll add a comment and an assert.